src/CommunityToolkit.Aspire.Hosting.Solr/SolrResource.cs

```diff
@@ -23,10 +23,36 @@ public class SolrResource(string name, string coreName) : ContainerResource(name
     /// </summary>
     public EndpointReference PrimaryEndpoint => _primaryEndpoint ??= new(this, PrimaryEndpointName);
 
+    /// <summary>
+    /// Gets the host endpoint reference for this resource.
+    /// </summary>
+    public EndpointReferenceExpression Host => PrimaryEndpoint.Property(EndpointProperty.Host);
+
+    /// <summary>
+    /// Gets the port endpoint reference for this resource.
+    /// </summary>
+    public EndpointReferenceExpression Port => PrimaryEndpoint.Property(EndpointProperty.Port);
+
     /// <summary>
     /// Gets the connection string expression for the Solr server.
     /// </summary>
     public ReferenceExpression ConnectionStringExpression =>  ReferenceExpression.Create(
             $"http://{PrimaryEndpoint.Property(EndpointProperty.Host)}:{PrimaryEndpoint.Property(EndpointProperty.Port)}/solr/{CoreName}");
```

Comment:
Shouldn't CoreName be encoded?

Reply:
What do you think a client would do with a URL? Split by '/' ? Also what if a client was actually reading the URL, this would break with any incompatible char then. This can always be changed later, not a big issue for now IMO